File: final-project/src/main/java/com/example/final_project/dto/CartItemResponse.java
package com.example.final_project.dto;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Builder
@Getter @Setter
public class CartItemResponse {
    private int id;  // cart_item.id
    private int productVariantId;  // ID của variant sản phẩm trong giỏ hàng
    private String productName;  // Tên sản phẩm
    private double price;  // Giá của sản phẩm
    private int quantity;  // Số lượng sản phẩm trong giỏ hàng
    private String image;  // Hình ảnh của sản phẩm (nếu có)
}

File: lib/models/CartItemResponse.dart
class CartItemResponse {
  final int id;
  final int productVariantId;
  final String productName;
  final String image;
  final int quantity;
  final double price;

  CartItemResponse({
    required this.id,
    required this.productVariantId,
    required this.productName,
    required this.image,
    required this.quantity,
    required this.price,
  });

  factory CartItemResponse.fromJson(Map<String, dynamic> json) {
    return CartItemResponse(
      id: json['id'] ?? 0, // hoặc ném lỗi nếu null
      productVariantId: json['productVariantId'] ?? 0,
      productName: json['productName'] ?? '',
      image: json['image'] ?? '',
      quantity: json['quantity'] ?? 0,
      price: (json['price'] as num?)?.toDouble() ?? 0.0,
    );
  }
}
